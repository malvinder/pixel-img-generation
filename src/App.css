.palette {
  display: flex;
  flex-direction: column;
  cursor: pointer;
  width: 256px;
  height: 128px;
}

.row {
  display: flex;
  flex-direction: row;
  height: 1px;
}

.col {
  width: 1px;
}
