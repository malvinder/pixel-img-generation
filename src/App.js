import React, { memo, useLayoutEffect, useRef } from 'react';
import './App.css';

const App = memo(() => {
  const height = 128;
  const width = 256;
  const canvasRef = useRef(null);

  const getColor = (red, green, blue) => {
    return `rgb(${red},${green},${blue})`;
  };

  const generateImage = () => {
    const canvas = canvasRef.current;
    const ctx = canvas.getContext('2d');

    for (let i=0; i<=255; i+=8) {
      for (let g=0; g<=255; g+=8) {
        for (let r=0; r<=255; r+=8) {
          ctx.fillStyle = getColor(i, r, g);
          ctx.fillRect(i, r, height, width);
        }
      }
    }
  };

  useLayoutEffect(() => {
    generateImage();
  }, []);
  return (
    <div className="palette">
      <canvas ref={canvasRef} width={width} height={height} />
    </div>
  );
});

export default App;
