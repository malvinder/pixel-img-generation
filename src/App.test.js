import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React, { useRef } from "react";
import App from "./App";

configure({ adapter: new Adapter() });

describe("rendering ColorPalette", () => {
  it("should render my canvas with colors", () => {
    const component = shallow(<App />);
    expect(component.getElements()).toMatchSnapshot();
  });

  it("should find the canvas element and match the given width and height", () => {
    const component = shallow(<App />);
    expect(component.find("canvas").props().width).toEqual(256);
    expect(component.find("canvas").props().height).toEqual(128);
  });
});
