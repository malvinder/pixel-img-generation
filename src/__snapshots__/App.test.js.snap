// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering ColorPalette should render my canvas with colors 1`] = `
Array [
  <div
    className="palette"
  >
    <canvas
      height={128}
      width={256}
    />
  </div>,
]
`;
